services/background-worker/src/Wire/BackendNotificationPusher.hs

```diff
@@ -339,18 +348,48 @@ startWorker rabbitmqOpts = do
       void $
         async $
           liftIO $
-            openConnectionWithRetries env.logger rabbitmqOpts $
+            openConnectionWithRetries env.logger "BackendNotificationPusher" rabbitmqOpts $ -- can we find a bug here, too?
               RabbitMqHooks
                 { -- The exception handling in `openConnectionWithRetries` won't open a new
                   -- connection on an explicit close call.
                   onNewChannel = \chan -> do
+                    LogNoClass.debug lgr $ Log.msg (Log.val "BackendNotificationPusher.startWorker: onNewChannel")
                     atomicWriteIORef chanRef $ pure chan
                     runAppT env $ startPusher client consumersRef chan,
                   onChannelException = \_ -> do
+                    LogNoClass.debug lgr $ Log.msg (Log.val "BackendNotificationPusher.startWorker: onNewException")
                     clearRefs
                     runAppT env $ markAsNotWorking BackendNotificationPusher,
                   onConnectionClose = do
+                    LogNoClass.debug lgr $ Log.msg (Log.val "BackendNotificationPusher.startWorker: onConnectionClose")
                     clearRefs
                     runAppT env $ markAsNotWorking BackendNotificationPusher
                 }
   pure (chanRef, consumersRef)
+
+-- cleanup will run in a new thread when the signal is caught, so we need to use IORefs and
+-- specific exception types to message threads to clean up
+cancelWorker :: WorkerResult -> AppT IO ()
+cancelWorker (notifChanRef, notifConsumersRef) = do
+  -- Notification pusher thread
+  Log.info $ Log.msg (Log.val "Cancelling the notification pusher thread")
+  readIORef notifChanRef >>= traverse_ \chan -> do
+    Log.info $ Log.msg (Log.val "Got channel")
```

Comment:
The channel's name might be interesting here as well.

Reply:
i can only find channelID, and that is only exported from Network.API.Internal, which is not exposed in the cabal file.  any hints where to find the name?